src/sys/signal.rs

```diff
@@ -488,102 +489,113 @@ pub enum SigevNotify {
                     si_value: libc::intptr_t },
 }
 
-/// Used to request asynchronous notification of the completion of certain
-/// events, such as POSIX AIO and timers.
-#[repr(C)]
-pub struct SigEvent {
-    sigevent: libc::sigevent
-}
+#[cfg(not(target_os = "openbsd"))]
+mod sigevent {
+    use libc;
+    use std::mem;
+    use std::ptr;
+    use std::fmt::{self, Debug};
+    use super::SigevNotify;
+    #[cfg(any(target_os = "linux", target_os = "freebsd"))]
+    use super::type_of_thread_id;
+
+    /// Used to request asynchronous notification of the completion of certain
+    /// events, such as POSIX AIO and timers.
+    #[repr(C)]
+    pub struct SigEvent {
+        sigevent: libc::sigevent
+    }
+
+    impl SigEvent {
+        // Note: this constructor does not allow the user to set the
```

Comment:
Can you turn this into a proper doc comment since you're modifying this? This is definitely something that should be exposed to our end users rather than only visible when browsing the source.